tracer/src/Datadog.Trace/SpanContextPropagator.cs

```diff
@@ -301,79 +234,47 @@ private static ulong ParseUInt64<T>(T carrier, Func<T, string, IEnumerable<strin
 
             if (hasValue)
             {
-                Log.Warning("Could not parse {HeaderName} headers: {HeaderValues}", headerName, string.Join(",", headerValues));
-            }
-
-            return 0;
-        }
-
-        private static SamplingPriority? ParseSamplingPriority<T>(T headers, string headerName)
-            where T : IHeadersCollection
-        {
-            var headerValues = headers.GetValues(headerName);
-
-            bool hasValue = false;
-
-            foreach (string headerValue in headerValues)
-            {
-                if (int.TryParse(headerValue, out var result))
-                {
-                    // note this int value may not be defined in the enum,
-                    // but we should pass it along without validation
-                    // for forward compatibility
-                    return (SamplingPriority)result;
-                }
-
-                hasValue = true;
-            }
-
-            if (hasValue)
-            {
-                Log.Warning(
-                    "Could not parse {HeaderName} headers: {HeaderValues}",
-                    headerName,
-                    string.Join(",", headerValues));
+                _log.Warning("Could not parse {HeaderName} headers: {HeaderValues}", headerName, string.Join(",", headerValues));
             }
 
-            return default;
+            return null;
         }
 
-        private static SamplingPriority? ParseSamplingPriority<T>(T carrier, Func<T, string, IEnumerable<string>> getter, string headerName)
+        private int? ParseInt32<TTCarrier>(TTCarrier carrier, Func<TTCarrier, string, IEnumerable<string?>> getter, string headerName)
```

Comment:
nit: Extra `T` character in this generic type

Reply:
oops, thanks!

Reply:
Fixed in 0767fcbcbdb7bace5ca893d8f47c058c8d244a04.